### PR TITLE
Add test for AuthenticationDetection

### PR DESCRIPTION
**Pull request by ChiuWang**
ref: [website issue](https://github.com/kubernetes/website/pull/8916)
Make  the feature fixed.
**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/authentication_test.go [hunk 1]
```diff
@@ -109,6 +109,28 @@ func TestAuthenticationDetection(t *testing.T) {
...
+			},
+			expected: rest.Config{BearerToken: "second"},
+		},
+		{
+			name:       "url with port2",
+			serverName: "one.com",
+			kubeconfig: clientcmdapi.Config{
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{
+					"one.com":     {Token: "first"},
+					"one.com:443": {Token: "second"},
```
Comment (CaoShuFeng) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/authentication_test.go]: nit: Put `one.com:443` in the first place. `one.com` in the second place.

**Review by CaoShuFeng — COMMENTED** [review]
/ok-to-test

**Comment by charrywanganthony** [comment]
ping @deads2k PTAL ref:[website #8916](https://github.com/kubernetes/website/pull/8916)

**Comment by charrywanganthony** [comment]
PTAL thanks. @deads2k 
/assign @sttts

**Pull request closed without merging**